cordapp-example/contracts-kotlin/src/main/kotlin/com/example/contract/CashContract.kt

```diff
@@ -18,14 +18,25 @@ class CashContract : Contract {
 
         when (command.value) {
             is Commands.Create -> requireThat {
-                // everyone can print/crate cash... because of covi... crisis
-                val cashState = tx.inputsOfType<CashState>().single()
-                "Cash value must be positive" using (cashState.balance > 0)
+                "There should be no input." using tx.inputsOfType<CashState>().isEmpty()
+                // in this exercise I allow anyone to create cash
+                val cashState = tx.outputsOfType<CashState>().single()
+                "Cash value must be positive" using (cashState.value > 0)
+                "Expect one signature" using (command.signers.size == 1)
+                "Creator must be a signer" using (command.signers.single() == cashState.owner.owningKey)
             }
             is Commands.Move -> requireThat {
-                // yet another dummy verification
-                val cashState = tx.inputsOfType<CashState>().first()
-                "Cash value must be positive" using (cashState.balance > 0)
+                "There should be one input." using (tx.inputsOfType<CashState>().size == 1)
+                "There should be one output." using (tx.outputsOfType<CashState>().size == 1)
+
+                val cashStateIn = tx.inputsOfType<CashState>().single()
+                val cashStateOut = tx.outputsOfType<CashState>().single()
+
+                "in/out value must match" using (cashStateIn.value.equals(cashStateOut.value))
+                "Can't move to the same party" using (!cashStateIn.owner.equals(cashStateOut.owner))
```

Comment:
Why not? Does it break the ledger? Or is it just that it sounds wasteful to you?

Reply:
sounds wasteful - why spamming with transactions that will not change a state at all?

Reply:
Wasteful, but does not break the ledger. The contract is only concerned with integrity. Plus, who knows, maybe when you want to redeem with the creating bank, maybe they don't accept small "coins", perhaps they want the redeem transactions to have few instances. So your contract should allow an owner to aggregate small amounts into larger ones.

Reply:
For aggregation I could create `Aggregate` command, right? and with aggregate I can add specific constraints like "owner must stay".
`Move` that doesn't move anything still looks not right for me.

Reply:
If the command was named 'Pay', it would sound even less appropriate, but 'Move' is fine. Move does not have information about the purpose. `Aggregate`, on the other hand, sounds excessive, to add a command only for that.